libtock/util/streaming_process_slice.c

```diff
@@ -16,34 +16,31 @@ returncode_t streaming_process_slice_init(
   streaming_process_slice_state_t* state,
   uint32_t                         driver,
   uint32_t                         allow,
-  void*                            buffer,
-  size_t                           size) {
-  // Each slice's header is 8 bytes long, and we create two slices from this
-  // buffer. Thus ensure that the provided buffer is at least 16 bytes long:
-  if (size < 16) {
+  void*                            buffer_a,
+  size_t                           size_a,
+  void*                            buffer_b,
+  size_t                           size_b) {
+  // Ensure that both buffers can hold the streaming process slice header:
+  if (size_a < STREAMING_PROCESS_SLICE_HEADER_LEN || size_b < STREAMING_PROCESS_SLICE_HEADER_LEN) {
     return RETURNCODE_ESIZE;
   }
 
   state->driver = driver;
   state->allow  = allow;
 
-  // We split the buffer in half, an application and kernel side. These two
-  // buffers are then atomically swapped with each other.
-  //
-  // Initially, the first half of this buffer is designated as the application
-  // buffer.
-  state->app_buffer_ptr  = buffer;
-  state->app_buffer_size = size / 2;
-
-  // Write a streaming process slice header to the second half of this buffer,
-  // and allow it to be the kernel buffer. We currently only support version
-  // 0, and don't set the `halt` flag:
-  uint8_t* kernel_buffer_ptr = state->app_buffer_ptr + state->app_buffer_size;
-  size_t kernel_buffer_size  = size - state->app_buffer_size;
-  streaming_process_slice_prepare_header(kernel_buffer_ptr);
+  // Initially, `buffer_a` is used as the application-owned buffer.
+  state->app_buffer_ptr     = buffer_a;
+  state->app_buffer_size    = size_a;
+  state->current_app_buffer = false; // false -> buffer_a
```

Comment:
We could rename this to `state->app_buffer_is_b` or something like that?

Reply:
Good idea.